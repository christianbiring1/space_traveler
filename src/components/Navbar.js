import React from 'react';
import { NavLink, Routes, Route } from 'react-router-dom';
import Profile from './Profile/Profile';
import Dragons from '../pages/Dragons';
import Mission from './Missions/Missions';
import Rockets from '../pages/Rockets';
import Logo from '../assets/planet.png';

function Navbar() {
  return (
    <>
      <div className="nav">
        <img src={Logo} alt="logo" className="logo" />
        <p className="logoTitle">Space Travelers Hub</p>
        <NavLink
          to="dragons"
          className={({ isActive }) => (isActive ? 'active' : 'inactive')}
        >
          Dragons
        </NavLink>
        <NavLink
          to="/missions"
          className={({ isActive }) => (isActive ? 'active' : 'inactive')}
        >
          Missions
        </NavLink>
        <NavLink
          to="/"
          className={({ isActive }) => (isActive ? 'active' : 'inactive')}
        >
          Rockets
        </NavLink>
        <NavLink
          to="/profile"
          className={({ isActive }) => (isActive ? 'active' : 'inactive')}
        >
          Profile
        </NavLink>
      </div>
      <Routes>
        <Route path="/dragons" element={<Dragons />} />
        <Route path="/missions" element={<Mission />} />
        <Route path="/" element={<Rockets />} />
        <Route path="/profile" element={<Profile />} />
      </Routes>
    </>
  );
}

export default Navbar;
