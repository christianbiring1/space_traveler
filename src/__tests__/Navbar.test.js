import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Navbar from '../components/Navbar';

it('Renders correctly', () => {
  const component = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
