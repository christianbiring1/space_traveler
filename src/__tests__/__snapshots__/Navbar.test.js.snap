// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
Array [
  <div
    className="nav"
  >
    <img
      alt="logo"
      className="logo"
      src="planet.png"
    />
    <p
      className="logoTitle"
    >
      Space Travelers Hub
    </p>
    <a
      className="inactive"
      href="/dragons"
      onClick={[Function]}
    >
      Dragons
    </a>
    <a
      className="inactive"
      href="/missions"
      onClick={[Function]}
    >
      Missions
    </a>
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
    >
      Rockets
    </a>
    <a
      className="inactive"
      href="/profile"
      onClick={[Function]}
    >
      Profile
    </a>
  </div>,
  <div
    data-testid="rocketsDisplay"
  />,
]
`;
